Grbl_Esp32/src/Serial.cpp

```diff
@@ -194,48 +191,28 @@ void serialCheckTask(void* pvParameters) {
     }
 }
 
-void serial_reset_read_buffer(uint8_t client) {
+void client_reset_read_buffer(uint8_t client) {
     for (uint8_t client_num = 0; client_num < CLIENT_COUNT; client_num++) {
         if (client == client_num || client == CLIENT_ALL) {
             client_buffer[client_num].begin();
         }
     }
 }
 
-// Writes one byte to the TX serial buffer. Called by main program.
-void serial_write(uint8_t data) {
-    Serial.write((char)data);
-}
-
-// Fetches the first byte in the serial read buffer. Called by protocol loop.
-uint8_t serial_read(uint8_t client) {
+// Fetches the first byte in the client read buffer. Called by protocol loop.
+int client_read(uint8_t client) {
     uint8_t data;
     vTaskEnterCritical(&myMutex);
     if (client_buffer[client].available()) {
         data = client_buffer[client].read();
         vTaskExitCritical(&myMutex);
```

Comment:
Not incorrect, but a bit hard to read that it's correct. I would make this single-entry-single-exit and an explicit cast: 

    int data = -1;
    vTaskEnterCritical(&myMutex);
    if (client_buffer[client].available()) {
        data = int(client_buffer[client].read());
    }
    vTaskExitCritical(&myMutex);
    return data;

Reply:
Yes your formulation is much better.  Changed it.

Reply:
It turns out that client_buffer  .read() already includes the -1 if not available logic, leading to a further simplification.